torch/distributed/_tensor/_collective_utils.py

```diff
@@ -158,3 +160,128 @@ def mesh_all_to_all(
             async_op=async_op,
         )
     return work
+
+
+def spec_to_bytes(spec: "placement_types.DTensorSpec") -> int:
+    assert spec.tensor_meta is not None, "spec should have tensor meta defined!"
+    return spec.tensor_meta.dtype.itemsize * math.prod(spec.shape)
+
+
+def get_bandwidth_factor(mesh: DeviceMesh) -> List[float]:
+    # generate bandwidth factor for intra-host/inter-host communication pattern
+    factors = [1.0 * mesh.ndim]
```

Comment:
Does this mean we are generating a list of one item or a list of mesh.ndim 1.0s?, IIUC, we want latter?

Reply:
good catch!!